components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -80,11 +81,11 @@ private void validate(String username, String userStoreDomain, String tenantDoma
         }
     }
 
+    // This method invokes only for the local users to resolve the user's unique user id from username.
     private List<String> getSessionsOfUser(String username, String userStoreDomain, String tenantDomain) throws
             UserSessionException {
 
-        String userId = UserSessionStore.getInstance().getUserId(username, getTenantId(tenantDomain),
-                userStoreDomain);
+        String userId = FrameworkUtils.resolveUserIdFromUsername(getTenantId(tenantDomain), userStoreDomain, username);
         return UserSessionStore.getInstance().getSessionId(userId);
```

Comment:
How this exactly should be fixed is,
- the legacy API method terminateSessionsOfUser(String username, String userStoreDomain, String tenantDomain) should consume the new API method terminateSessionsByUserId(String userId)
- therefore, within terminateSessionsOfUser(String username, String userStoreDomain, String tenantDomain) you will resolve the user id invoking FrameworkUtils.resolveUserIdFromUsername(getTenantId(tenantDomain), userStoreDomain, username) and then invoke terminateSessionsByUserId(String userId)
- So, eventually when federated user case comes into picture FrameworkUtils.resolveUserIdFromUsername(getTenantId(tenantDomain), userStoreDomain, username should be able to resolve the user id for such users.
- This make the private method which you have used redundant. so you can remove it actually